pandas/io/formats/html.py

```diff
@@ -491,6 +491,15 @@ class NotebookFormatter(HTMLFormatter):
     DataFrame._repr_html_() and DataFrame.to_html(notebook=True)
     """
 
+    def __init__(self, formatter, classes=None, border=None):
```

Comment:
does this really need to be in ``__init__``, rather have this call a method in the base class which is overriden here, e.g.

``self.columns = self._get_columns_formatted_values()``

Reply:
> rather have this call a method in the base class which is overriden here

in #24651, i separated out the notebook functionality from `HTMLFormatter` into `NotebookFormatter` using `HTMLFormatter` as the base class. the base class of `HTMLFormatter` is `TableFormatter` which is also the base class of `DataFrameFormatter` (used for `to_string` and `to _latex`) and `LatexFormatter`.

I envisaged that at some point in the development cycle it may become desirable to also create a `ToHTMLFormatter` using `HTMLFormatter` as the base.

IMO `HTMLFormatter` and `NotebookFormatter` are not the appropriate location for non-markup related formatting issues that are common across output-formatting methods. I think this issue is in that category and should ideally be in `DataFrameFormatter` or `TableFormatter`.

However, if to close the open issue the fix is somewhere in `io/formats/html.py`, then i think a  TODO to remove the code at a later date should be added.

Reply:
>> and the bug was present in `to_html(notebook=False)` as well
>
> I would argue it is not a bug here. since the display options should not affect the to_html() output. compare with the display options, `max_rows`, `max_columns`, `show_dimensions`, `max_colwidth`, etc. which only apply to the notebook display.

I think I'm getting a bit confused here. Doesn't this mean that this is not an issue that is common across formatters? Because `to_html` should only check display preferences when used for display purposes in a notebook and not when generating HTML for a web page?

Reply:
yeah. i can understand why your confused, there is a slight problem with the current class hierarchy. the workaround for `max_colwidth` was to use `with option_context` to ignore the display option for `to_html`, see #24841

Reply:
i think the ideal solution would be to have say a `use_display_options` attribute in `DataFrameFormatter` and then in `HTMLFormatter` we just have `self.fmt.use_display_options = False` and in `NoteBookFormatter` we have `self.fmt.use_display_options = True` and then all the display formatting could be handled in `io/formats/format.py`.

But this is way outside the scope of this PR. So a `with option_context` work-around would be fine for now IMO.